Classes/game.py

```diff
@@ -279,19 +279,33 @@ def buildItems(self):
         with open("Items/defaultLocations.json") as f:
             itemDict = json.load(f)
         f.close()
-        for key, value in itemDict.items():
-            itemName = key + ".json"
-            file = os.path.join("Items", itemName)
-            if type(value) == list:
+
+        # Open the item's file
+        for itemName, value in itemDict.items():
+            itemPath = itemName + ".json"
+            file = os.path.join("Items", itemPath)
+
+            # Set location and visibility
+            location = value["location"]
+            visible = False
+            if value["visible"].lower() == "true":
+                visible = True
+
+            # Place the items
+            if type(location) == list:
                 item = Item(file)
                 for room in self.rooms:
-                    if room == value[0]:
-                        room.addItem(item)
-                value = value[1]
+                    if room == location[0] and visible:
+                        room.addVisibleItem(item)
+                    elif room == location[0] and not visible:
+                        room.addHiddenItem(item)
+                location = location[1]
             item = Item(file)
             for room in self.rooms:
-                if room == value:
-                    room.addItem(item)
+                if room == location and visible:
+                    room.addVisibleItem(item)
+                elif room == location and not visible:
+                    room.addHiddenItem(item)
 
```

Comment:
I'm not sure what you mean. There are two calls to addVisibleItem and addHiddenItem, but one pair handles items in multiple locations and the other handles items in just one location. Items are showing only one instance in my own testing, and this follows the format of the original function, with just the item categories changed. While this chunk of code could probably be refactored to be more concise, that can be a separate PR. Can you clarify where the issue is?

Reply:
![image](https://user-images.githubusercontent.com/61394328/201497056-fae3b671-06e7-442a-bdd9-83d89d909de0.png)

If location is a list, then both the code in the green bracket AND code in the blue bracket will execute. Did you mean to put the blue bracket in an `else:` block? Then either green bracket OR blue bracket would execute.

Reply:
The green block already only executes when the item has multiple locations, not for every item. Like this:

- If itemA has more than one location (LocationA, LocationB)...
     - Duplicate the item (create itemB)
     - Put itemB where it belongs
     - Change itemA from a location list to a single location
- Check itemA's location
     - Put itemA in the appropriate place

Reply:
Ok I think I understand what's happening now.

If there is more than one location, then green creates and adds the item for location[0] only. Then blue would create and add for location[1]. 

Otherwise, blue creates and adds for the only location.

Reply:
Exactly. The green only handles the multiplied items. The blue handles the first instance of all items.

Reply:
Would it be better overall design to have location just be the same data type (list) no matter what? If location is always a list, then just iterate over the list and perform the create and add operations for each location.

It would definitely make this easier to understand. Would also make the code scalable, because this implementation only supports either 1 location or max 2 locations.

Reply:
Alternatively, we could have the green block execute for locations with index > 0.

Then the blue block would always execute for the first location (either location[0] or when location is not a list)

Reply:
Let me know if you want to consider those approaches, or if you'd rather just keep it as-is I can approve and merge.

Reply:
Yes, I agree that it could be refactored, but Charlie wrote it the way it works at the moment, and I'd have to ask if there was a reason he went with this solution. It will take more time to figure that out, and I'd rather make that another PR later.

Reply:
Good point! My apologies, I lost sight of the fact this approach was already how it was begin done previously. Wish I would have made those suggestions whenever it got approved before, rather than bringing it up now. Thanks!